src/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -379,7 +379,7 @@ internal bool WaitForExit(int millisecondsTimeout)
                 // We're in a polling loop... determine how much time remains
                 int remainingTimeout = millisecondsTimeout == Timeout.Infinite ?
                     Timeout.Infinite :
-                    (int)Math.Max(millisecondsTimeout - (GetTimestamp() - startTime), 0);
+                    (int)Math.Max(millisecondsTimeout - ((Stopwatch.GetTimestamp() - startTime) / (double)Stopwatch.Frequency * 1000), 0);
```

Comment:
It could, but then a Stopwatch would need to be allocated.  If the code were any more complex, it'd probably be worthwhile, but as it stands, seems better to just do this manually and avoid the unnecessary allocation.